### ISSUE TITLE
DiagnosticContext doesn't use the logger created by LoggerConfiguration

### ISSUE THREAD

**Issue by cbersch** [serilog/serilog-extensions-hosting#52]
When using the 
```c#
public static IHostBuilder UseSerilog(
    this IHostBuilder builder,
    Action<HostBuilderContext, IServiceProvider, LoggerConfiguration> configureLogger,
    bool preserveStaticLogger = false,
    bool writeToProviders = false)
```
extension and calling it with `preserveStaticLogger: true`, the `ILogger`, which is created by `LoggerConfiguration`, is not used by `DiagnosticContext`.
I would expect, the logger instance itself to be always passed to DiagnosticContext.

**Comment by nblumhardt** [comment]
Interesting - this sounds like it would be a bug, needs some 👀 - thanks!

**Comment by cbersch** [comment]
Ok, I gave it a try and fixed it in [#53](https://github.com/serilog/serilog-extensions-hosting/pull/53).

**Comment by fvarukhin** [comment]
I faced with the same issue with using Lamar and Serilog, - standard Serilog initialization doesn't works with Lamar initialization.
I was failed to find solution in internet and started to serf in source code of Serilog and Lamar
I want to suggest work around which is works in my case.

1. You need to create Serilog configuration class

```csharp
public class SerilogConfiguration : ServiceRegistry
    {
        public SerilogConfiguration(IConfiguration config)
        {
            Log.Logger = new LoggerConfiguration()
                .ReadFrom.Configuration(config)
                .CreateLogger();
            this.AddLogging(loggingBuilder =>
                loggingBuilder.AddSerilog());
            DiagnosticContext implementationInstance = new DiagnosticContext(null);
            this.AddSingleton(implementationInstance);
            this.AddSingleton((IDiagnosticContext) implementationInstance);
        }
    }
```

2. In your program.cs use this code for Serilog initialization in my case Im using appSettings.json for storing serilog configuration

```csharp
public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
                .ConfigureLogging((context, logging) => logging.ClearProviders())
                .UseLamar((builder,registry) =>
                {
                    registry.IncludeRegistry(new SerilogConfiguration(builder.Configuration));
                })
                .ConfigureWebHostDefaults(webBuilder => { webBuilder.UseStartup<Startup>(); });
```
